lib/irb/completion.rb

```diff
@@ -93,6 +93,29 @@ def retrieve_files_to_require_relative_from_current_dir
     end
   end
 
+  class TypeCompletor < BaseCompletor # :nodoc:
+    def initialize(source_file = nil)
+      require 'repl_type_completor'
+      ReplTypeCompletor.preload_rbs
+      @source_file = source_file
```

Comment:
Does this mean in the entire IRB session, user only gets the completion targeting one source file? For example, if I put a `binding.irb` under `app/models/user.rb`, will it run all analysis only around `user.rb`?

Reply:
Nice catch. Actually, this initialization  (build_type_completor and also `IRB::Irb.new`) will be called in each `binding.irb`.
But I found that `@irb_path` is not fully initialized yet when build_completor is called and `context.irb_path` is overwritten after `IRB::Irb` initialization.
```ruby
# In irb.rb:1050
binding_irb = IRB::Irb.new(workspace)
binding_irb.context.irb_path = irb_path
binding_irb.run(IRB.conf)
```

I will change to `IRB::TypeCompletor.new(context)` and call `ReplTypeCompletor.analyze(preposing + target, binding: bind, filename: context.irb_path)` on completion phase.

Reply:
Yeah I think that's better because with `irb:rdbg`, `irb_path` could be updated if user moves around with the debugger. If we store it as a state in InputMethod, it won't be synced correctly.